modules/blob-storing-database/src/Volo.Abp.BlobStoring.Database.Domain/Volo/Abp/BlobStoring/Database/DatabaseBlobNameCalculator.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Volo.Abp.DependencyInjection;
+using Volo.Abp.MultiTenancy;
+
+namespace Volo.Abp.BlobStoring.Database
+{
+    public class DatabaseBlobNameCalculator : IDatabaseBlobNameCalculator, ITransientDependency
+    {
+        protected ICurrentTenant CurrentTenant { get; }
+
+        public DatabaseBlobNameCalculator(ICurrentTenant currentTenant)
+        {
+            CurrentTenant = currentTenant;
+        }
+
+        public virtual string Calculate(BlobProviderArgs args)
+        {
+            if (!args.Configuration.IsMultiTenant)
+            {
+                return args.BlobName;
+            }
+
+            return CurrentTenant.Id == null
+                ? $"host/{args.BlobName}"
+                : $"tenants/{CurrentTenant.Id.Value.ToString("D")}/{args.BlobName}";
```

Comment:
@hikalkan 

We can also add a property to the `DatabaseBlob` instead of change the `Name`. eg  `Shared`.

Before:
-|TenantId|Name
-----|-----|-----
Host|NULL|host/123.jpg
Tenant|1|tenant/123.jpg
Not Multi Tenant|NULL|123.jpg

After:
-|TenantId|Shared|Name
-----|-----|-----|-----
Host|NULL|FALSE|123.jpg
Tenant|1|FALSE|123.jpg
Not Multi Tenant|NULL|TRUE|123.jpg